### PR TITLE
New structure for code examples that follow modules names

### PR DESCRIPTION
**Pull request by mlachkar**

### CODE REVIEW

File: 05_loops/src/main/scala/more_loop_examples/Tasks.scala [hunk 1]
```diff
@@ -1,4 +1,3 @@
-package org.epfl.tasks
```
Comment (julienrf) [inline 05_loops/src/main/scala/more_loop_examples/Tasks.scala:1]: Is there a reason to remove this package clause? It is bad practice to write definitions in the “empty” package, so I prefer to keep a package, if that’s OK?

File: build.sbt [hunk 2]
```diff
@@ -1,16 +1,54 @@
...
+
+lazy val testing =
+  project.in(file("10_testing"))
+    .enablePlugins(WorksheetsRunner)
+
+lazy val typeDirectedProgramming =
+  project.in(file("11_type_directed_programming"))
+    .enablePlugins(WorksheetsRunner)
+
+lazy val ExtensionMethodsAndImplicitConversions =
```
Comment (julienrf) [inline build.sbt]: Minor detail, but do you mind using the same casing convention for all project names? (start with a lower case letter)

File: week_2/src/main/scala/org/epfl/13_sierpinski/Sierpinski.scala [hunk 3]
```diff
@@ -1,65 +0,0 @@
...
-      unit
-    else
-      val previous = loop(n - 1)
-      previous.above(
-        previous.beside(previous)
-      )
-
-  val image = loop(iterations)
-  image.draw()
-end recursion
```
Comment (julienrf) [inline week_2/src/main/scala/org/epfl/13_sierpinski/Sierpinski.scala:65]: I would like to keep this file. It belongs to the “loops” module.

File: 14_asynchronous_programming/build.sbt [hunk 4]
```diff
@@ -0,0 +1 @@
+scalaVersion := "3.1.1"
```
Comment (julienrf) [inline 14_asynchronous_programming/build.sbt:1]: This project does not have a `project/build.properties` file.

File: 09_reasoning_about_code/js/src/main/scala/reasonnng_about_the_code/Canvas.scala [hunk 5]
```diff
@@ -1,4 +1,4 @@
-package canvas
+package reasonnng_about_the_code
```
Comment (julienrf) [inline 09_reasoning_about_code/js/src/main/scala/reasonnng_about_the_code/Canvas.scala]:
```suggestion
package reasoning_about_code
```
(the directory name should be fixed as well)

File: 09_reasoning_about_code/build.sbt [hunk 6]
```diff
@@ -0,0 +1,10 @@
+scalaVersion := "3.1.1"
+
+val root =
+  crossProject(JSPlatform, JVMPlatform)
+    .jsSettings(
+      libraryDependencies ++=
+        Seq(
+          ("org.scala-js" %%% "scalajs-dom" % "1.1.0").cross(CrossVersion.for3Use2_13),
+        )
+    )
```
Comment (julienrf) [inline 09_reasoning_about_code/build.sbt]: Previously, there was a file `index.html` to see the program output, it seems to be missing now?

**Review by julienrf — COMMENTED** [review]
Thank you for improving the structure!

I noticed a few typos, see my comments. Also, could you please restore the individual README files for each module?

File: 05_loops/src/main/scala/more_loop_examples/Tasks.scala (hunk 1 above)
Comment (mlachkar) [reply 05_loops/src/main/scala/more_loop_examples/Tasks.scala:1]: it's intellij which remove it. but I will add the correct package where it was deleted

File: build.sbt (hunk 2 above)
Comment (mlachkar) [reply build.sbt]: of course

File: week_2/src/main/scala/org/epfl/13_sierpinski/Sierpinski.scala (hunk 3 above)
Comment (mlachkar) [reply week_2/src/main/scala/org/epfl/13_sierpinski/Sierpinski.scala:65]: do you know where it's supposed to be ? 
It's a mistake.

Comment (mlachkar) [reply week_2/src/main/scala/org/epfl/13_sierpinski/Sierpinski.scala:65]: ok

File: build.sbt (hunk 2 above)
Comment (mlachkar) [reply build.sbt]: done!

File: week_2/src/main/scala/org/epfl/13_sierpinski/Sierpinski.scala (hunk 3 above)
Comment (mlachkar) [reply week_2/src/main/scala/org/epfl/13_sierpinski/Sierpinski.scala:65]: done

File: 14_asynchronous_programming/build.sbt (hunk 4 above)
Comment (mlachkar) [reply 14_asynchronous_programming/build.sbt:1]: done

File: 09_reasoning_about_code/build.sbt (hunk 6 above)
Comment (mlachkar) [reply 09_reasoning_about_code/build.sbt]: done

File: 05_loops/src/main/scala/more_loop_examples/Tasks.scala (hunk 1 above)
Comment (mlachkar) [reply 05_loops/src/main/scala/more_loop_examples/Tasks.scala:1]: it's fixed

File: 05_loops/src/main/scala/loops/Sierpinski.scala [hunk 7]
```diff
@@ -41,7 +41,7 @@ end collections
   ((n - 1) to 0 by -1).foreach { i =>
     (0 until (n - i)).foreach { j =>
       if (j & i) == 0 then
-      images += unit.at((i / 2.0 + j) * resolution, i * resolution.toDouble)
+        images += unit.at((i / 2.0 + j) * resolution, i * resolution.toDouble)
```
Comment (julienrf) [inline 05_loops/src/main/scala/loops/Sierpinski.scala:44]: :eyes: good catch!

File: 08_modules/README.md [hunk 8]
```diff
@@ -0,0 +1,9 @@
+Invoke the following command to generate the images of the houses:
+
+~~~ shell
+sbt run
+~~~
+
+This will create a file houses.pdf.
```
Comment (julienrf) [inline 08_modules/README.md]: I am not sure this README is the correct one? The program that creates the houses comes earlier in the curriculum IIRC (not in module 8)

File: 11_type_directed_programming/src/main/scala/type_classes/type_classes_01_generic_ordering.worksheet.sc [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+case class Rational(num: Int, denom: Int)
+
+given Ordering[Rational] with
+  def compare(r1: Rational, r2: Rational): Int =
+    r1.num * r2.denom - r2.num * r1.denom
+
+val rationals = List(Rational(6, 2), Rational(7,2), Rational(5,2), Rational(5, 3))
+
+sort(rationals)
+sort(rationals)(using Ordering[Rational].reverse)
```
Comment (julienrf) [inline 11_type_directed_programming/src/main/scala/type_classes/type_classes_01_generic_ordering.worksheet.sc]: Does this file duplicate something? Or was it missing in the first place? (Git marks it as an added file, not a moved file)

File: 11_type_directed_programming/src/main/scala/type_classes/type_classes_02_subtyping.worksheet.sc [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+case class Rational(num: Int, denom: Int) extends Comparable:
+  def compareTo(that: Comparable): Int = that match
+    case other: Rational =>
+      num * other.denom - other.num * denom
+    case _ =>
+      throw new IllegalArgumentException
+  
+Rational(5, 3).compareTo(Rational(2, 1))
+Rational(6, 3).compareTo(Rational(2, 1))
+Rational(7, 3).compareTo(Rational(2, 1))
```
Comment (julienrf) [inline 11_type_directed_programming/src/main/scala/type_classes/type_classes_02_subtyping.worksheet.sc]: Same about this file

File: README.md [hunk 11]
```diff
@@ -16,24 +16,21 @@ The code samples fall in two categories:
...
 # Extracting code for a specific week
 
 The `extract` script allows one to create a _zip_ archive that
 contains just the code for one of the course weeks.
 
-It requires to arguments, the first being the week number and the
+It requires to arguments, the first being the module name and the
 second, a folder in which the _zip_ archive will be created.
 
+> for example: sh extract 02_basics /tmp
```
Comment (julienrf) [inline README.md]:
```suggestion
For example: `sh extract 02_basics /tmp`

```

**Review by julienrf — APPROVED** [review]
Awesome, thank you Meriam!

Could you please enable the CI on PRs before we merge it? You can do that as a separate commit on your branch.

File: 08_modules/README.md (hunk 8 above)
Comment (mlachkar) [reply 08_modules/README.md]: you're right!

File: README.md (hunk 11 above)
Comment (mlachkar) [reply README.md]: ok!

File: 11_type_directed_programming/src/main/scala/type_classes/type_classes_01_generic_ordering.worksheet.sc (hunk 9 above)
Comment (mlachkar) [reply 11_type_directed_programming/src/main/scala/type_classes/type_classes_01_generic_ordering.worksheet.sc]: it's a duplicate! thanks!

File: 11_type_directed_programming/src/main/scala/type_classes/type_classes_02_subtyping.worksheet.sc (hunk 10 above)
Comment (mlachkar) [reply 11_type_directed_programming/src/main/scala/type_classes/type_classes_02_subtyping.worksheet.sc]: duplicate too!

**Pull request merged**